### PR TITLE
Port Answerbox functionality to TypeScript

### LINKED ISSUE #160: Port Answerbox functionality to TypeScript
In an effort to refactor the frontend portion of the site, we want to move from CoffeeScript to TypeScript and in the very same step also drop stuff like jQuery to the bare minimum.

This is the sub-issue of the refactor for the answerbox functionality.

**To Do:**
* [ ] Opening the comment section
* [ ] Adding a new comment
* [ ] Removing comments
* [ ] (Un)smiling comments
* [ ] Reporting comments
* [ ] Comment character count
* [ ] Removing an answer
* [ ] Reporting an answer
* [ ] (Un)smiling an answer
* [ ] (Un)subscribing to answer updates

### PR DESCRIPTION
**Pull request by pixeldesu**
Fixes #160 

**Testing:**
* Opening the comment section
* Adding a new comment
* Removing comments
* (Un)smiling comments
* Reporting comments
* Comment character count
* Removing an answer
* Reporting an answer
* (Un)smiling an answer
* (Un)subscribing to answer updates

### CODE REVIEW

**Comment by raccube** [comment]
Bug: Navigating between pages seems to break the comment toggle

**Comment by raccube** [comment]
Bug: Actions trigger multiple times after navigating between pages (both answers & comments)

(Observed on Smile button & comment posting)
image
image

**Comment by raccube** [comment]
General: lack of animation when opening comments doesn't feel right

**Review by raccube — CHANGES_REQUESTED** [review]
Noticed a fair chunk of turbolinks bugs

**Comment by pixeldesu** [comment]
@raccube Turbolinks issues should be fixed.

We talked about why the animations are removed at the moment, there needs to be more setup to bring them back because there is just no good replacement for all the slide-methods.

File: app/javascript/retrospring/features/answerbox/comment/new.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      success: (data) => {
+        if (data.success) {
+          document.querySelector(`#ab-comments-${id}`).innerHTML = data.render;
+          document.querySelector(`#ab-comment-count-${id}`).innerHTML = data.count;
+          input.value = '';
+          counter.innerHTML = String(160);
+
+          const sub = document.querySelector<HTMLElement>(`[data-action=ab-submarine][data-a-id="${id}"]`);
+          sub.dataset.torpedo = "no"
+          sub.children[0].nextSibling.textContent = I18n.translate('views.actions.unsubscribe');
```
Comment (raccube) [inline app/javascript/retrospring/features/answerbox/comment/new.ts]:
```suggestion
          sub.children[0].nextSibling.textContent = ' ' + I18n.translate('views.actions.unsubscribe');
```

**Review by raccube — CHANGES_REQUESTED** [review]

File: app/javascript/retrospring/features/answerbox/subscribe.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  Rails.ajax({
+    url: targetUrl,
+    type: 'POST',
+    data: new URLSearchParams({
+      answer: id
+    }).toString(),
+    success: (data) => {
+      if (data.success) {
+        button.dataset.torpedo = ["yes", "no"][torpedo];
+        button.children[0].nextSibling.textContent = torpedo ? I18n.translate('views.actions.unsubscribe') : I18n.translate('views.actions.subscribe');
```
Comment (raccube) [inline app/javascript/retrospring/features/answerbox/subscribe.ts]: Also missed this one
```suggestion
        button.children[0].nextSibling.textContent = ' ' + (torpedo ? I18n.translate('views.actions.unsubscribe') : I18n.translate('views.actions.subscribe'));
```

**Review by raccube — CHANGES_REQUESTED** [review]

**Review by raccube — APPROVED** [review]

**Pull request merged**